modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/DefaultOperationContext.java

```diff
@@ -153,4 +171,50 @@ public RuntimeOperationModel getOperationModel()
     {
         return operationModel;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public MuleContext getMuleContext()
+    {
+        return muleContext;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public synchronized Optional<TransactionConfig> getTransactionConfig()
```

Comment:
Move the synchronization inside the block where transactionConfig is initialized